### PR TITLE
Reflect the latest resolved image on Deployers

### PR DESCRIPTION
**Pull request by scothis**
This means that the Deployer's status will update when the target build
has a new latest image. The Adapter resource already follows this
pattern.

### CODE REVIEW

File: pkg/controllers/core/deployer_controller.go [hunk 1]
```diff
@@ -284,6 +284,9 @@ func (r *DeployerReconciler) constructDeploymentForDeployer(deployer *corev1alph
 			},
 		},
 	}
+	if deployment.Spec.Template.Spec.Containers[0].Image == "" {
```
Comment (sbawaska) [inline pkg/controllers/core/deployer_controller.go]: The deployment will not pick up the latest image if one was set previously. This is a change from the previous behavior. remove the if condition?

Comment (scothis) [reply pkg/controllers/core/deployer_controller.go]: We previously used this field as a holder for the resolved image, but it was never persisted back to the api server. If the container image is defined, then it wins.

Previously we validated that either an image or build ref was specified, but not both. We should bring that back.

Comment (scothis) [reply pkg/controllers/core/deployer_controller.go]: one thing we could do is surface the value of the PodSpec container as lastestImage when there is no build.

**Comment by sbawaska** [comment]
Thanks! Ship it!

**Pull request merged**